export/mlab_export.py

```diff
@@ -0,0 +1,591 @@
+  M-Lab.
+
+Overview of default operation:
+  * Script finds all RRD files under --rrddir_prefix.
+  * Script opens a file for output under --output_dir (optionally compressed).
+  * Script reads values from each RRD between --ts_start and --ts_end.
+  * Script determines ts_start and ts_end from 'mtime' on LAST_EXPORT_FILENAME.
+    - On first run, LAST_EXPORT_FILENAME is created. ts_end is set to nearest
+      hour from current time, and ts_start is set to ts_end minus --length.
+    - On later runs, ts_start is set to the 'mtime' of LAST_EXPORT_FILENAME.
+      And, ts_end is set to the most recent hour.
+  * Script only exports metrics defined in the export_metrics.conf config file.
+  * On success, script sets mtime on LAST_EXPORT_FILENAME to ts_end, in
+    preparation for the next run.
+
+Examples:
+  # Default operation should not require additional parameters.
+  ./mlab-export.py
+
+  # Useful for testing, --noupdate exports the last hour without the
+  # side-effects of modifying the mtime of LAST_EXPORT_FILENAME.
+  ./mlab-export.py --noupdate --output example.json
+
+  # To export a different set of metrics than the global default.
+  ./mlab-export.py --noupdate --output example.json --export_metrics metrics.cfg
+
+  # List rrd file names, the raw metric names, or the canonical metric names.
+  ./mlab-export.py --show_rrdfile
+  ./mlab-export.py --show_rawmetric --show_metric
+
+  # Show the collectd-nagios command line for checking each metric.
+  ./mlab-export.py --show_nagios
+"""
+import ConfigParser
+import fcntl
+import gzip
+import json
+import logging
+import os
+import socket
+import sys
+import time
+
+# Third-party modules.
+import gflags as flags
+import rrdtool
+
+
+COLLECTD_INTERVAL = 10
+EXPORT_DIR = '/var/spool/mlab_utility'
+HOSTNAME = None
+LAST_EXPORT_FILENAME = '/var/lib/collectd/lastexport.tstamp'
+LIST_OPTIONS = ('rrdfile', 'metric', 'metric_raw', 'metric_skipped')
+METRIC_MAP = None
+METRIC_MAP_CONF = '/usr/share/collectd-mlab/export_metrics.conf'
+RRD_PREFIX = '/var/lib/collectd/rrd/'
+
+flags.DEFINE_string(
+    'rrddir_prefix', RRD_PREFIX, 'Root directory of RRD files to export.')
+flags.DEFINE_integer(
+    'length', 3600, 'Length of time to export in seconds.', lower_bound=0)
+flags.DEFINE_integer(
+    'step', COLLECTD_INTERVAL, 'Time between RRD values in seconds. This value '
+    'must equal the value in the collectd config. Inaccurate values will not '
+    'change the intervals of exported data, but could result in samples being '
+    'skipped at the beginning or end of the export window.', lower_bound=1)
+flags.DEFINE_integer(
+    'ts_start', None,
+    'Timestamp to start export, in seconds since the epoch.', lower_bound=0)
+flags.DEFINE_integer(
+    'ts_end', None,
+    'Timestamp to end export, in seconds since the epoch.', lower_bound=0)
+flags.DEFINE_integer(
+    'ts_offset', 900, 'Amount of time (seconds) that must have past after '
```

Comment:
Typo "passed" fixed. 42ed366a02b533c887983299554c53462a5eb82d

Also changed ts_offset default value to 600 here, as discussed in #9.